build/management/commands/build_rfaa_models.py

```diff
@@ -0,0 +1,1013 @@
+                os.remove('./{}.pdb'.format(sd['pdb']))
+            except KeyError:
+                print("////////////////////////////WARNING: {} GN assign failed".format(sd['pdb']))
+
+            pdbdata, created = PdbData.objects.get_or_create(pdb=pdbdata_raw)
+            struct.pdb_data = pdbdata
+
+            self.parsed_pdb = PDBParser(PERMISSIVE=True, QUIET=True, get_header=True).get_structure('ref', pdb_path)[0]
+
+            if 'pdb' in sd:
+                web_resource = WebResource.objects.get(slug='pdb')
+                struct.pdb_code, created = WebLink.objects.get_or_create(index=sd['pdb'], web_resource=web_resource)
+            else:
+                self.logger.error('PDB code not specified for structure {}, skipping!'.format(sd['pdb']))
+                continue
+
+            # structure type
+            sd['structure_method'] = sd['model']
+            if 'structure_method' in sd and sd['structure_method']:
+                structure_type = 'Model (RFAA)'
+                structure_type_slug = slugify(sd['structure_method'])
+                try:
+                    st, created = StructureType.objects.get_or_create(slug=structure_type_slug, defaults={'name': structure_type})
+                    if created:
+                        self.logger.info('Created structure type {}'.format(st))
+                except IntegrityError:
+                    st = StructureType.objects.get(slug=structure_type_slug)
+                struct.structure_type = st
+            else:
+                self.logger.warning('No structure type specified in PDB file {}'.format(sd['pdb']))
+
+#################################################
+
+            # insert into plain text fields
+            if 'preferred_chain' in sd:
+                struct.preferred_chain = sd['preferred_chain']
+            else:
+                self.logger.warning('Preferred chain not specified for structure {}'.format(sd['pdb']))
+            if 'resolution' in sd:
+                struct.resolution = float(sd['resolution'])
+            else:
+                struct.resolution = None
+                self.logger.warning('Resolution not specified for structure {}. Setting as null'.format(sd['pdb']))
+            if 'publication_date' in sd:
+                struct.publication_date = sd['publication_date']
+            else:
+                sd['publication_date'] = build_date = date.today()
+                struct.publication_date = sd['publication_date']
+                self.logger.warning('Publication date not specified for structure {}. Defaulting at today'.format(sd['pdb']))
+
+            struct.annotated = True
+            struct.refined = False
+            struct.stats_text = None
+
+            #Resolution and
+            # save structure before adding M2M relations
+            struct.save()
+
+            print('BEFORE CREATING LIGAND')
+
+            # Create LigandPeptideStructure object
```

Comment:
these are not peptide ligands so I guess this shouldn't be here? If you want to include small molecule ligand objects it should be with StructureLigandInteraction